pool/limiter.go

```diff
@@ -91,5 +91,10 @@ func (r *RateLimiter) withinLimit(ip string, clientType int) bool {
 		// create a new limiter if the incoming request is from a new client.
 		reqLimiter = r.addRequestLimiter(ip, clientType)
 	}
+
+	if reqLimiter == nil {
```

Comment:
I'm unsure how I feel about this. I think the code would be cleaner if `addRequestLimiter()` didn't return nil when it is passed an unknown client type, but instead returned an error.

To my eye, the way this code is currently written looks more like Java than Go.